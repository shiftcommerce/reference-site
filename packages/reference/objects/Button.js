// Libraries
import { Component } from 'react'
import classNames from 'classnames'

class Button extends Component {
  render () {
    let {
      status,
      label,
      size,
      disabled,
      className,
      ...otherProps
    } = this.props

    disabled = disabled || false
    status = status || 'primary'
    size = size || 'lrg'

    return (
      <button
        className={classNames(className, 'o-button', `o-button--${status}`, `o-button--${size}`, {'o-button--disabled': disabled})}
        tabIndex='0'
        role='button'
        disabled={disabled}
        {...otherProps}
      >
        { label }
      </button>
    )
  }
}

export default Button
