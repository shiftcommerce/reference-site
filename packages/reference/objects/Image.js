import { Component } from 'react'
import classNames from 'classnames'

class Image extends Component {
  styles () {
    let basicStyles = {
      height: this.props.height,
      width: this.props.width
    }

    if (basicStyles.height < 200) {
      return { ...basicStyles, backgroundSize: ((basicStyles.height * 19) / 20) }
    } else if (basicStyles.width < 200) {
      return { ...basicStyles, backgroundSize: ((basicStyles.width * 19) / 20) }
    } else {
      return { ...basicStyles, backgroundSize: 150 }
    }
  }

  render () {
    let {
      src,
      className
    } = this.props

    if (src) {
      return <img {...this.props} />
    } else {
      return <div className={classNames('c-dummy_image', className)} style={this.styles()} />
    }
  }
}

export default Image
