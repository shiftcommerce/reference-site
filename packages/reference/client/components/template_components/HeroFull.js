// Libraries
import { Component } from 'react'
import Link from 'next/link'

// Objects
import Button from '../../objects/Button'
import Image from '../../objects/Image'

class HeroFull extends Component {
  heroHeading (hero) {
    // Only render if heading available
    if (hero.heading) {
      return (
        <div className='c-hero__heading'>{hero.heading}</div>
      )
    }
  }

  heroOverlay (hero, currentPosition) {
    let overlayPosition = hero.overlay_position[0] ? hero.overlay_position[0] : 'below'
    if (this.shouldDisplayOverlay(hero, overlayPosition, currentPosition)) {
      return (
        <div className='c-hero__overlay' style={{ backgroundColor: hero.overlay_colour }}>
          {this.overlayPointer(hero.overlay_colour, overlayPosition, 'above')}
          <div className='c-hero__overlay_content'>
            <div className='c-hero__overlay_title'>
              {hero.title}
            </div>
            {hero.text}
            {this.heroButtons(hero)}
          </div>
          {this.overlayPointer(hero.overlay_colour, overlayPosition, 'below')}
        </div>
      )
    }
  }

  shouldDisplayOverlay (hero, overlayPosition, currentPosition) {
    // only render if the current position matches the component position
    const overlayPositionConditional = (overlayPosition === currentPosition)
    // Should display overlay if title, text or a link available
    const overlayDataConditional = (hero.title || hero.text || (hero.link_1_url && hero.link_1_text))

    return overlayPositionConditional && overlayDataConditional
  }

  overlayPointer (colour, overlayPosition, pointerPosition) {
    // only render if the current position matches the component position
    if (overlayPosition !== pointerPosition) {
      let style = {}
      if (pointerPosition === 'below') {
        style['borderTopColor'] = colour
      } else {
        style['borderBottomColor'] = colour
      }
      return <div className={`c-hero__overlay_pointer c-hero__overlay_pointer--${pointerPosition}`} style={style} />
    }
  }

  heroButtons (hero) {
    let buttons = []
    let index = 1

    // Only start looping if first button is there
    if (hero.link_1_url && hero.link_1_text) {
      while (hero[`link_${index}_url`] && hero[`link_${index}_text`]) {
        buttons.push(
          <div className='c-hero__button' key={index}>
            <Link href={`/slug?slug=${hero[`link_${index}_url`][0].canonical_path}`} as={hero[`link_${index}_url`][0].canonical_path}>
              <Button className='c-hero__button-icon' label={hero[`link_${index}_text`]} size='lrg' aria-label={hero[`link_${index}_text`]} />
            </Link>
          </div>
        )
        index++
      }
    }

    return buttons
  }

  render () {
    const { componentData } = this.props

    return (
      <div className='c-hero c-hero--full_with_gutters' style={{ backgroundColor: componentData.background_colour }}>
        {this.heroHeading(componentData)}
        <div className='c-hero__content'>
          {this.heroOverlay(componentData, 'above')}
          <Image className='c-hero__image' src={componentData.image[0].s3_url} />
          {this.heroOverlay(componentData, 'below')}
        </div>
      </div>
    )
  }
}

export default HeroFull
