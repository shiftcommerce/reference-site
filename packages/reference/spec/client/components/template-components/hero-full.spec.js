// Components
import HeroFull from '../../../../client/components/template-components/hero-full'

// Objects
import Button from '../../../../client/objects/button'
import Image from '../../../../client/objects/image'

// Fixtures
import heroImageData from '../../../fixtures/hero-full'

test('renders the full HeroImage component', () => {
  // Arrange
  const emptyFunction = () => {}
  const image = heroImageData.image[0].s3_url
  const mobileImage = heroImageData.mobile_image && heroImageData.mobile_image[0] && heroImageData.mobile_image[0].s3_url
  const linkText = heroImageData.overlay_link_1_text

  // Act
  const wrapper = mount(
    <HeroFull componentData={heroImageData} />
  )

  // Assert
  expect(wrapper).toMatchSnapshot()
  expect(wrapper).toContainReact(<Image className='c-hero__image' src={image} mobileSrc={mobileImage} />)
  expect(wrapper.find('Button')).toMatchElement(<Button className='c-hero__button-icon' label={linkText} status='warning' size='lrg' aria-label={linkText} onClick={emptyFunction} />)
})
