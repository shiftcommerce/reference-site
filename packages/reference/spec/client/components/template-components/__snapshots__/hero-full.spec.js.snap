// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the full HeroImage component 1`] = `
<HeroFull
  componentData={
    Object {
      "background_colour": "antiquewhite",
      "heading": "",
      "id": "4_1",
      "image": Array [
        Object {
          "asset_file": "1513596567.2856915-Week-44-Department-Homeware-2_24.jpg",
          "canonical_path": "https://shift-platform-dev-assets.s3-eu-west-1.amazonaws.com/uploads/asset_file/asset_file/157604/1513596567.2856915-Week-44-Department-Homeware-2_24.jpg",
          "created_at": "2017-12-18T11:29:27Z",
          "file_content_content_type": "image/jpeg",
          "file_content_filename": "Week-44-Department-Homeware-2_24.jpg",
          "file_content_size": 354806,
          "id": "157604",
          "image_height": 1483,
          "image_width": 2396,
          "reference": "26418193-2571-4bb8-943b-bb567065c4f8",
          "s3_url": "https://shift-platform-dev-assets.s3-eu-west-1.amazonaws.com/uploads/asset_file/asset_file/157604/1513596567.2856915-Week-44-Department-Homeware-2_24.jpg",
          "updated_at": "2017-12-18T11:29:27Z",
        },
      ],
      "mobile_image": null,
      "overlay_background_colour": "antiquewhite",
      "overlay_image_overlap": Array [
        "medium",
      ],
      "overlay_link_1_text": "Test Product",
      "overlay_link_1_url": Array [
        Object {
          "canonical_path": "/homeware/dining/glassware",
          "id": "/homeware/dining/glassware",
          "path": "/homeware/dining/glassware",
          "url": "/homeware/dining/glassware",
        },
      ],
      "overlay_link_2_text": "",
      "overlay_link_2_url": null,
      "overlay_link_3_text": "",
      "overlay_link_3_url": null,
      "overlay_link_4_text": "",
      "overlay_link_4_url": null,
      "overlay_link_5_text": "",
      "overlay_link_5_url": null,
      "overlay_link_background_colour": "aquamarine",
      "overlay_link_border_colour": "mediumseagreen",
      "overlay_link_hover_background_colour": "black",
      "overlay_link_hover_border_colour": "white",
      "overlay_link_hover_text_colour": "white",
      "overlay_link_text_colour": "goldenrod",
      "overlay_position": Array [
        "below",
      ],
      "overlay_text": "Words Words Words",
      "overlay_text_colour": "blueviolet",
      "overlay_title": "Title",
      "reference": "hero_full",
      "type": "components",
    }
  }
>
  <section
    className="o-template-component o-template-component--full-width c-hero u-padding-top-des"
    style={
      Object {
        "backgroundColor": "antiquewhite",
      }
    }
  >
    <div
      className="c-hero__content"
    >
      <ConditionalLink
        href="/homeware/dining/glassware"
      >
        <Link
          as="/homeware/dining/glassware"
          href="/slug?slug=/homeware/dining/glassware"
        >
          <a
            href="/homeware/dining/glassware"
            onClick={[Function]}
          >
            <Image
              className="c-hero__image"
              src="https://shift-platform-dev-assets.s3-eu-west-1.amazonaws.com/uploads/asset_file/asset_file/157604/1513596567.2856915-Week-44-Department-Homeware-2_24.jpg"
            >
              <div
                className="c-hero__image"
              >
                <picture>
                  <img
                    className="c-image"
                    src="https://shift-platform-dev-assets.s3-eu-west-1.amazonaws.com/uploads/asset_file/asset_file/157604/1513596567.2856915-Week-44-Department-Homeware-2_24.jpg"
                    srcSet="https://shift-platform-dev-assets.s3-eu-west-1.amazonaws.com/uploads/asset_file/asset_file/157604/1513596567.2856915-Week-44-Department-Homeware-2_24.jpg 769w"
                  />
                </picture>
              </div>
            </Image>
          </a>
        </Link>
      </ConditionalLink>
      <div
        className="c-hero__overlay c-hero__overlay--overlap_below_medium c-hero__overlay--offset_below_medium_4"
        style={
          Object {
            "backgroundColor": "antiquewhite",
            "color": "blueviolet",
          }
        }
      >
        <div
          className="c-hero__overlay_pointer c-hero__overlay_pointer--above"
          style={
            Object {
              "borderBottomColor": "antiquewhite",
            }
          }
        />
        <div
          className="c-hero__overlay_content"
        >
          <div
            className="c-hero__overlay_title"
          >
            Title
          </div>
          <div
            className="u-whitespace_preline"
          >
            Words Words Words
          </div>
          <div
            className="c-hero__button"
            key="1"
          >
            <Link
              as="/homeware/dining/glassware"
              href="/slug?slug=/homeware/dining/glassware"
            >
              <a
                className="c-hero__button-icon"
                href="/homeware/dining/glassware"
                onClick={[Function]}
              >
                Test Product
              </a>
            </Link>
          </div>
        </div>
      </div>
    </div>
  </section>
</HeroFull>
`;
